src/middleware/gzip_middleware.py

```
@@ -0,0 +1,327 @@
+            else:
+                if self.enable_metrics:
+                    self.DECOMPRESSION_ERRORS.labels(endpoint=path, error_type="size_limit").inc()
+                logger.error(f"Decompression size limit exceeded for {path}: {e}")
+
+                if self.fail_on_error:
+                    await self._send_error_response(send, _HTTP_413, error_message)
+                else:
+                    raw_body = compressed_body if compressed_body is not None else b"".join(body_parts)
+                    await self._passthrough_with_body(scope, raw_body, send)
+
+        except gzip.BadGzipFile as e:
```

Flake8 (7.3.0)

[error] local variable 'e' is assigned to but never used

(F841)